src/EnergyPlus/ReportCoilSelection.cc

```diff
@@ -979,6 +905,65 @@ int ReportCoilSelection::getIndexForOrCreateDataObjFromCoilName(EnergyPlusData &
     return index;
 }
 
+void ReportCoilSelection::associateZoneCoilWithParent(EnergyPlusData &state, std::unique_ptr<CoilSelectionData> &c)
+{
+    c->coilLocation = "Unknown";
+    c->typeHVACname = "Unknown";
+    c->userNameforHVACsystem = "Unknown";
+    // now search equipment
+    auto const &zoneEquipList = state.dataZoneEquip->ZoneEquipList(c->zoneEqNum);
+    bool keepLooking = true;
+    for (int equipLoop = 1; equipLoop <= zoneEquipList.NumOfEquipTypes; ++equipLoop) {
+        if (zoneEquipList.EquipName(equipLoop) == c->coilName_) {
+            c->typeHVACname = zoneEquipList.EquipTypeName(equipLoop);
+            c->userNameforHVACsystem = zoneEquipList.EquipName(equipLoop);
+            c->coilLocation = "Zone Equipment";
+            keepLooking = false;
+            break;
+        }
+        if (keepLooking) {
+            if (zoneEquipList.EquipData(equipLoop).Name == c->coilName_) {
+                c->typeHVACname = zoneEquipList.EquipTypeName(equipLoop);
+                c->userNameforHVACsystem = zoneEquipList.EquipName(equipLoop);
+                c->coilLocation = "Zone Equipment";
+                keepLooking = false;
+                break;
+            }
+            if (keepLooking) {
+                for (int subEq = 1; subEq <= zoneEquipList.EquipData(equipLoop).NumSubEquip; ++subEq) {
+                    if (zoneEquipList.EquipData(equipLoop).SubEquipData(subEq).Name == c->coilName_) {
+                        c->typeHVACname = zoneEquipList.EquipTypeName(equipLoop);
+                        c->userNameforHVACsystem = zoneEquipList.EquipName(equipLoop);
+                        c->coilLocation = "Zone Equipment";
```

Comment:
These three lines are the same for sub and sub sub, so they could be a single block at the end 
`if (!keepLooking){ do these}`
This could also be setting `c->zoneName` and `c->zoneNum` (Zone Name is still showing N/A in the details table).